script/newslideshow.js

```diff
@@ -0,0 +1,41 @@
+import { fetchData } from "./common.js";
+
+//Henter ut data fra json-fil ved hjelp av funksjon i common
+async function getData() {
+  const data = await fetchData();
+
+  //Kaller på funksjon til å bytte bilde hvert 5. sekund
+  let counter = 1;
+  setTimeout(
+    () =>
+      setInterval(() => {
+        console.log(counter);
+        swapImg(data[counter]);
+        counter++;
+        if (counter == data.length) {
+          counter = 0;
+        }
+      }, 5000),
+    800
+  );
+}
+
+function swapImg(data) {
+  let slideshow = document.getElementById("imgSlide");
```

Comment:
Burde denne være global eller skal jeg definere den i hver funksjon som jeg gjør nå

Reply:
Man vil holde seg unna globale variabler, så det er en bedre løsning å definere den hver gang. En enda mer optimal løsning hadde jo ikke hatt samme kode flere ganger, det man kunne gjort er:

definere variabelen `slideshow` øverst i `getData()` funksjonen. så passer du `slideshow` som et argument til `swapImg` funksjonen og enda en gang som argument til `fadeIn` funksjonen. Da blir både funksjonene mere beskrivende `fadeIn(slideshow)` kontra `fadeIn()` (hva fades inn??) og man definere bare slideshow på et sted 🚀 

Samme når du kaller `fadeOut` (som burde flyttes inn i `getData`), pass `slideshow` som et argument. Syntaxen for å sende inn parameter på måten `fadeOut` blir blir da:
```
setInterval(() => fadeOut(slideshow), 5000)
```